File: app/src/main/java/me/eycia/api/PicRef.java
package me.eycia.api;

import android.os.Parcel;
import android.os.Parcelable;
import android.util.Pair;

/**
 * Created by eycia on 3/3/16.
 */
public class PicRef implements Parcelable{
    public String Url;
    public String Ref;
    public int px, py;
    public String Description;

    public PicRef(String description, Pair<Integer, Integer> pixes, String ref, String url) {
        Description = description;
        px = pixes.first;
        py = pixes.second;
        Ref = ref;
        Url = url;
    }

    protected PicRef(Parcel in) {
        Url = in.readString();
        Ref = in.readString();
        px = in.readInt();
        py = in.readInt();
        Description = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(Url);
        dest.writeString(Ref);
        dest.writeInt(px);
        dest.writeInt(py);
        dest.writeString(Description);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<PicRef> CREATOR = new Creator<PicRef>() {
        @Override
        public PicRef createFromParcel(Parcel in) {
            return new PicRef(in);
        }

        @Override
        public PicRef[] newArray(int size) {
            return new PicRef[size];
        }
    };
}

File: app/src/main/java/me/eycia/api/PicRef.kt
package me.eycia.api

import android.os.Parcel
import android.os.Parcelable

/**
 * Created by eycia on 3/3/16.
 */
data class PicRef(var Url: String,
                  var Ref: String,
                  var px: Int = 0,
                  var py: Int = 0,
                  var Description: String) : Parcelable {

    constructor(`in`: Parcel) : this(
            Url = `in`.readString(),
            Ref = `in`.readString(),
            px = `in`.readInt(),
            py = `in`.readInt(),
            Description = `in`.readString()) {
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(Url)
        dest.writeString(Ref)
        dest.writeInt(px)
        dest.writeInt(py)
        dest.writeString(Description)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {

        @JvmField final val CREATOR: Parcelable.Creator<PicRef> = object : Parcelable.Creator<PicRef> {
            override fun createFromParcel(`in`: Parcel): PicRef {
                return PicRef(`in`)
            }

            override fun newArray(size: Int): Array<PicRef?> {
                return arrayOfNulls(size)
            }
        }
    }
}
